### PR TITLE
Fixed spelling and grammatical errors.

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: docs/proposals/OptimizerEffects.rst [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 .. OptimizerEffects:
 
-Optimizer Effects: Summarizing and specifing function side effects
+Optimizer Effects: Summarizing and specifying function side effects
 ==================================================================
```
Comment (gribozavr) [inline docs/proposals/OptimizerEffects.rst]: Please adjust the underline to match the header length.

Comment (MrWaknis) [inline docs/proposals/OptimizerEffects.rst]: I did it, just now.

**Comment by krememek** [comment]
The diff still shows the underline is not of the same length as the text above it.

**Comment by gribozavr** [comment]
Seems like these issues have been fixed already.

**Pull request closed without merging**